crates/ruff_linter/src/rules/pylint/rules/access_member_before_definition.rs

```diff
@@ -0,0 +1,102 @@
+use ruff_diagnostics::{Diagnostic, FixAvailability, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_semantic::{ScopeId, ScopeKind};
+
+use ruff_source_file::SourceRow;
+use ruff_text_size::Ranged;
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for accesses on instance attributes before they are initialized in __init__ method.
+/// If the attribute is not initialized at all then it is not reported.
+///
+/// ## Why is this bad?
+/// Accessing a class member before it is initialized will raise an `AttributeError`.
+///
+/// ## Example
+/// ```python
+/// class Unicorn:
+///     def __init__(self, x):
+///         self.x = x
+///         if self.x > 9000:  # [access-member-before-definition]
+///             pass
```

Comment:
I think the example here is incorrect
```suggestion
///     def __init__(self, x):
///         if self.x > 9000:  # [access-member-before-definition]
///             pass
///         self.x = x
```